odoo/addons/base/models/ir_asset.py

```diff
@@ -0,0 +1,362 @@
+        a specific asset and target the right bundle, i.e. the first one
+        defining the target path.
+
+        :param target_path_def: string: path to match.
+        :root_bundle: string: bundle from which to initiate the search.
+        :returns: the first matching bundle or None
+        """
+        ext = target_path_def.split('.')[-1]
+        installed = self._get_installed_addons_list()
+        target_path = self._get_paths(target_path_def, installed)[1][0]
+
+        css = ext in STYLE_EXTENSIONS
+        js = ext in SCRIPT_EXTENSIONS
+        xml = ext in TEMPLATE_EXTENSIONS
+
+        asset_paths = self.get_asset_paths(root_bundle, css=css, js=js, xml=xml)
+
+        for path, _, bundle in asset_paths:
+            if path == target_path:
+                return bundle
+
+        return root_bundle
+
+    def _get_active_addons_list(self):
+        """Can be overridden to filter the returned list of active modules."""
+        return self._get_installed_addons_list()
+
+    def _get_asset_domain(self, bundle):
+        """Meant to be overridden to add additional parts to the search domain"""
+        return [('bundle', '=', bundle), ('active', '=', True)]
+
+    def _get_installed_addons_list(self):
+        """
+        Returns the list of all installed addons.
+        :returns: string[]: list of module names
+        """
+        if not http.request:
+            return self.env['ir.module.module'].sudo()._installed_sorted()
+        else:
+            return http.module_boot()
+
+    def _get_manifest_cache(self):
+        """Proxy to the http addons manifest, used for test overrides."""
+        return http.addons_manifest
+
+    def _get_paths(self, path_def, installed, extensions=None):
+        """
+        Returns a list of file paths matching a given glob (path_def) as well as
+        the addon targetted by the path definition. If no file matches that glob,
+        the path definition is returned as is. This is either because the glob is
+        not correctly written or because it points to an URL.
+
+        :param path_def: the definition (glob) of file paths to match
+        :param installed: the list of installed addons
+        :param extensions: a list of extensions that found files must match
+        :returns: a tuple: the addon targetted by the path definition [0] and the
+            list of glob files matching the definition [1] (or the glob itself if
+            none). Note that these paths are filtered on the given `extensions`.
+        """
+        paths = []
+        path_url = fs2web(path_def)
+        path_parts = [part for part in path_url.split('/') if part]
+        addon = path_parts[0]
+        addon_manifest = self._get_manifest_cache().get(addon)
+
+        if addon_manifest:
+            if addon not in installed:
+                raise Exception("Unallowed to fetch files from addon %s" % addon)
+            addons_path = os.path.join(addon_manifest['addons_path'], '')[:-1]
```

Comment:
Did you try to attack and get /etc/passwd.js ?

Reply:
yes, see test_31 to 36